modules/auth/proc-adding-google-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,59 @@
+[id='proc-adding-google-as-an-authentication-provider_{context}']
+= Enabling the Google authentication provider
+
+{product} includes a Google authentication provider that can authenticate users by using Google OAuth2.
+
+.Procedure
+. To allow {product-short} to authenticate with Google, create OAuth credentials in Google.
+For more information, see the official Google Cloud documentation for link:https://developers.google.com/identity/openid-connect/openid-connect#appsetup[Setting up OAuth 2.0].
++
+.. To authorize {product-short} to access the user consent screen, enter the necessary information on the https://console.cloud.google.com/apis/credentials/consent/edit[*Google Cloud > Oauth consent screen > Edit app registration*] page, including the following:
++
+... On the *OAuth consent screen* tab, enter your {product-short} domain (_<APP_FQDN>_) into the *Authorized domains* field.
+... On the *Scopes* tab, select the following scopes:
++
+openid:: Associate you with your personal info on Google.
+auth/userinfo.email:: See your primary Google Account email address.
+auth/userinfo.profile:: See your personal info, including any personal info you've made publicly available.
+... On the *Test users* tab, add yourself as a test user, if using the External user type.
+
+.. To create your OAuth *Client ID* and *Client secret*, on the link:https://console.cloud.google.com/apis/credentials[*APIs & Services > Credentials*] page, click *Create Credentials*, choose *OAuth client ID*, and enter the necessary informations including the following:
+... In the *Application type* dropdown, select `Web Application`.
+... In the *Authorized JavaScript origins* section, enter the {product-short} URI: `pass:c,a,q[https://_<APP_FQDN>_]`.
+... In the *Authorized redirect URIs* section, enter the backend authentication URI set in {product-short}:
+`pass:c,a,q[https://_<APP_FQDN>_/api/auth/google/handler/frame]`.
+... Save the *Client ID* and *Client secret* for the next step.
+
+. Add your Google credentials in your {product-short} secrets.
++
+Edit your {product-short} secrets, such as `secrets-rhdh`.
++
+Add the following key/value pairs:
++
+`AUTH_GOOGLE_CLIENT_ID`:: Enter the *Client ID* that you generated on Google.
+`AUTH_GOOGLE_CLIENT_SECRET`:: Enter the *Client secret* that you generated on Google.
+
+. Set up the Google authentication provider in your {product-short} custom configuration.
++
+Edit your custom {product-short} config map, such as `app-config-rhdh`.
++
+In the `app-config-rhdh.yaml` content, add the `google` provider configuration to your `app-config.yaml` under the root `auth` configuration, and enable the `google` provider for sign-in:
```

Comment:
These are the steps that seem to be missing numbers. And the key/value pairs might need bullets.